backend/src/controllers/auth-controller.ts

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable-next-line padded-blocks */
+import bcrypt from "bcrypt";
+import { Request, Response } from "express";
+import dotenv from "dotenv";
+import jwt, { JwtPayload, VerifyErrors } from "jsonwebtoken";
+import {
+  findUserByEmail,
+  registerNewUser,
+} from "../repositories/user-repository";
+import { createWallet } from "../repositories/wallet-repositry";
+import { generateReferralCode } from "../utils/generate-referral-code"
+
+dotenv.config();
+
+export const handleLogin = async (req: Request, res: Response) => {
```

Comment:
bad naming. no need for `handle` prefix. consider it's user's controller and this can be just login. others too. change them.